src/http/gateway/routes/gateway.js

```diff
@@ -2,19 +2,37 @@
 
 const resources = require('../resources')
 
-module.exports = {
-  method: '*',
-  path: '/ipfs/{cid*}',
-  options: {
-    handler: resources.gateway.handler,
-    pre: [
-      { method: resources.gateway.checkCID, assign: 'args' }
-    ],
-    response: {
-      ranges: false // disable built-in support, we do it manually
-    },
-    ext: {
-      onPostHandler: { method: resources.gateway.afterHandler }
+module.exports = [
+  {
+    method: '*',
+    path: '/ipfs/{immutableId*}',
```

Comment:
i think we should keep this as CID or else we are just making up another concept and we already have enough

Reply:
I looked closer and this path is actually cid+optional path, so I renamed it to `cidPath` (name already used in [one of is-ipfs checks](https://www.npmjs.com/package/is-ipfs#isipfscidpathpath)).

Reply:
what about using `ipfs path` and `ipns path` both represent more than cid/keys

so a string representation of this type `/ipfs/<something>*` would be ipfs path, same for ipns 


CID is an object with a string representation as a baseX of the multihash.

Reply:
To keep things simple I renamed both to `ipfsPath` and `ipnsPath` respectively: https://github.com/ipfs/js-ipfs/pull/2020/commits/eebde0bc8591f63066e665d65f81c8646a15e083
Maybe it should be just `path` in both? I really want to remove custom names where possible.

Reply:
yes we should just call this `path` 

ipfsPath would be the complete request.path `'/ipfs/{path*}'`